swagger2.0/oic.r.rule.swagger.json

```diff
@@ -0,0 +1,371 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OCF Rule",
+    "version": "20190910",
+    "license": {
+      "name": "OCF Data Model License",
+      "url": "https://github.com/openconnectivityfoundation/core/blob/e28a9e0a92e17042ba3e83661e4c0fbce8bdc4ba/LICENSE.md",
+      "x-copyright": "copyright 2019 Open Connectivity Foundation, Inc. All rights reserved."
+    },
+    "termsOfService": "https://openconnectivityfoundation.github.io/core/DISCLAIMER.md"
+  },
+  "schemes": ["http"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/RuleResURI?if=oic.if.ll": {
+      "get": {
+        "description": "A Rule is a Collection made up of 3 Links. \n A Link to an instance of a Collection of Rule Inputs. \n A Link to a Rule Expression (the logic of the Rule). \n A Link to a Collection of Rule Actions.",
+        "parameters": [
+          {"$ref": "#/parameters/interface-all"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "Retrieves the rule as Links List.",
+            "x-example":
+            [
+              {
+                "href": "/ruleinputcollection",
+                "rt":   ["oic.r.rule.inputcollection"],
+                "if":   ["oic.if.ll","oic.if.baseline"],
+                "p":    {"bm": 3},
+                "eps": [
+                    {"ep": "coaps://[fe80::b1d6]:1111"}
+                ]
+              },
+              {
+                "href": "/ruleexpression",
+                "rt":   ["oic.r.rule.expression"],
+                "if":   ["oic.if.rw","oic.if.baseline"],
+                "p":    {"bm": 3},
+                "eps": [
+                    {"ep": "coaps://[fe80::b1d6]:1111"}
+                ]
+              },
+              {
+                "href": "/ruleactioncollection",
+                "rt":   ["oic.wk.rule.actioncollection"],
+                "if":   ["oic.if.ll","oic.if.baseline"],
+                "p":    {"bm": 3},
+                "eps": [
+                    {"ep": "coap://[fe80::b1d6]:1111"}
+                ]
+              }
+            ],
+            "schema": { "$ref": "#/definitions/slinklist" }
+          }
+        }
+      }
+    },
+    "/RuleResURI?if=oic.if.baseline": {
+      "get": {
+        "description": "A Rule is a Collection made up of 3 Links: \n- A Link to an instance of a Collection of Rule Inputs. \n- A Link to a Rule Expression (the logic of the Rule). \n- A Link to a Collection of Rule Actions.",
+        "parameters": [
+          {"$ref": "#/parameters/interface-all"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "Retrieves the baseline response for the rule.",
+            "x-example":
+            {
+              "rt": ["oic.r.rule"],
+              "if": ["oic.if.ll","oic.if.baseline","oic.if.create"],
```

Comment:
The use of "oic.if.create" for this Resource is not defined in the CR.  We need to update the CR to include how this is used.  Also, are new Resources actually created?  Or are we just adding Links to existing Resources (i.e. oic.if.linkadd)?

Reply:
Is there a use case for oic.if.b support?

Reply:
The CR discusses what can be created and what is a linkadd.  There is no case for batch as of now.